packages/google_adsense/lib/src/ad_unit_widget_web.dart

```diff
@@ -0,0 +1,201 @@
+
+  @override
+  State<AdUnitWidgetWeb> createState() => _AdUnitWidgetWebState();
+}
+
+class _AdUnitWidgetWebState extends State<AdUnitWidgetWeb>
+    with AutomaticKeepAliveClientMixin {
+  static int adUnitCounter = 0;
+  double adHeight = 1.0;
+  late web.HTMLElement adUnitDiv;
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return SizedBox(
+      height: adHeight,
+      child: HtmlElementView.fromTagName(
+          tagName: 'div', onElementCreated: onElementCreated),
+    );
+  }
+
+  void onElementCreated(Object element) {
+    adUnitDiv = element as web.HTMLElement;
+    log('onElementCreated: $adUnitDiv with style height=${element.offsetHeight} and width=${element.offsetWidth}');
+    adUnitDiv
+      ..id = 'adUnit${adUnitCounter++}'
+      ..style.height = 'min-content'
+      ..style.textAlign = 'center';
+    // Adding ins inside of the adUnit
+    adUnitDiv.append(widget._insElement);
+
+    // TODO(sokoloff06): Make shared
+    // Using Resize observer to detect element attached to DOM
+    web.ResizeObserver((JSArray<web.ResizeObserverEntry> entries,
+            web.ResizeObserver observer) {
+      for (final web.ResizeObserverEntry entry in entries.toDart) {
+        final web.Element target = entry.target;
+        if (target.isConnected) {
+          // First time resized since attached to DOM -> attachment callback from Flutter docs by David
+          if (!(target as web.HTMLElement)
+              .dataset
+              .getProperty('attached' as JSString)
+              .isDefinedAndNotNull) {
+            onElementAttached(target);
+            target.dataset
+                .setProperty('attached' as JSString, true as JSBoolean);
+            observer.disconnect();
+          }
+        }
+      }
+    }.toJS)
+        .observe(adUnitDiv);
+
+    // Using Mutation Observer to detect when adslot is being loaded based on https://support.google.com/adsense/answer/10762946?hl=en
+    web.MutationObserver(
+            (JSArray<JSObject> entries, web.MutationObserver observer) {
+      for (final JSObject entry in entries.toDart) {
+        final web.HTMLElement target =
+            (entry as web.MutationRecord).target as web.HTMLElement;
+        log('MO current entry: $target');
+        if (isLoaded(target)) {
+          observer.disconnect();
+          if (isFilled(target)) {
+            updateWidgetHeight(target.offsetHeight);
+          } else {
+            // Prevent scrolling issues over empty ad slot
+            target.style.pointerEvents = 'none';
+            target.style.height = '0px';
+            updateWidgetHeight(0);
+          }
+        }
+      }
+    }.toJS)
+        .observe(
+            widget._insElement,
+            web.MutationObserverInit(
+                attributes: true,
+                attributeFilter:
+                    <String>['data-ad-status'].jsify()! as JSArray<JSString>));
```

Comment:
This line is a bit odd, you shouldn't need all these conversions or the as cast I think!